src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/AvoidSingleUseOfLocalJsonSerializerOptions.cs

```diff
@@ -0,0 +1,328 @@
+
+            if (jsonSerializerOptionsSymbol is null || jsonSerializerSymbol is null)
+            {
+                return;
+            }
+
+            context.RegisterOperationAction(
+                context =>
+                {
+                    var operation = (IObjectCreationOperation)context.Operation;
+
+                    INamedTypeSymbol? typeSymbol = operation.Constructor?.ContainingType;
+                    if (SymbolEqualityComparer.Default.Equals(typeSymbol, jsonSerializerOptionsSymbol))
+                    {
+                        if (IsCtorUsedAsArgumentForJsonSerializer(operation, jsonSerializerSymbol) ||
+                            IsLocalUsedAsArgumentForJsonSerializerOnly(operation, jsonSerializerSymbol))
+                        {
+                            context.ReportDiagnostic(operation.CreateDiagnostic(s_Rule));
+                        }
+                    }
+                },
+                OperationKind.ObjectCreation);
+        }
+
+        private static bool IsCtorUsedAsArgumentForJsonSerializer(IObjectCreationOperation objCreationOperation, INamedTypeSymbol jsonSerializerSymbol)
+        {
+            IOperation operation = WalkUpConditional(objCreationOperation);
+
+            return operation.Parent is IArgumentOperation argumentOperation &&
+                IsArgumentForJsonSerializer(argumentOperation, jsonSerializerSymbol);
+        }
+
+        private static bool IsArgumentForJsonSerializer(IArgumentOperation argumentOperation, INamedTypeSymbol jsonSerializerSymbol)
+        {
+            return argumentOperation.Parent is IInvocationOperation invocationOperation &&
+                SymbolEqualityComparer.Default.Equals(invocationOperation.TargetMethod.ContainingType, jsonSerializerSymbol);
+        }
+
+        private static bool IsLocalUsedAsArgumentForJsonSerializerOnly(IObjectCreationOperation objCreation, INamedTypeSymbol jsonSerializerSymbol)
+        {
+            IOperation operation = WalkUpConditional(objCreation);
+
+            if (!IsLocalAssignment(operation, out List<ILocalSymbol>? localSymbols))
+            {
+                return false;
+            }
+
+            IBlockOperation? localBlock = objCreation.GetFirstParentBlock();
+            bool ret = false;
+
+            foreach (ILocalReferenceOperation descendant in localBlock.Descendants().OfType<ILocalReferenceOperation>())
+            {
+                if (!localSymbols.Contains(descendant.Local))
+                {
+                    continue;
+                }
+
+                // Avoid cases that would potentially make the local escape current block scope.
+                if (IsArgumentOfJsonSerializer(descendant, jsonSerializerSymbol, out bool isArgumentOfInvocation))
+                {
+                    // Case: used more than once i.e: not single-use.
+                    if (ret)
```

Comment:
Nit: consider using a more descriptive name for `ret`, e.g. `isJsonSerializerInvocationFound`.